### PR TITLE
chore: Clean up eslint configuration for packages

### PR DESCRIPTION
**Pull request by scinos**
Continuation of #56746

#### Changes proposed in this Pull Request

* Change packages' `eslintrc.js` config to extend from `@automattic/calypso-build/eslint/node` (where the package is meant to be run by Node).
* Delete `//eslint-disable-next-line no-console` and similar that got superseded by the changes in the package's `eslintrc.js`
* Refactor some `//eslint-disable-line no-console` to `//eslint-disable-next-line no-console`

#### Testing instructions

N/A

### CODE REVIEW

**Comment by scinos** [comment]
This is causing lots of yarn warnings because every package that pulls in `@automattic/calypso-build` needs to also pull in all the peer dependencies: webpack, jest, enzyme...

I think is overkill, a better approach is to split `@automattic/calypso-build` into smaller packages. In this case, `@automattic/calypso-build-eslint` with just the eslint config (#56818)

This PR is blocked until #56818 is merged.

**Comment by scinos** [comment]
This is ready to review now.

File: packages/calypso-config/src/index.ts
Comment (WunderBart) [inline packages/calypso-config/src/index.ts]: Just a tiny nit - noticed that we're [not consistent](https://github.com/Automattic/wp-calypso/pull/56787/files#diff-0ee25e9e72d92852c2f17dd888a8da40a3788ecb5a5e71a88243ec383fd5fd9cR12) with the whitespace after `//`.

**Review by WunderBart — APPROVED** [review]
Looks good! 🚀

File: packages/calypso-config/src/index.ts
Comment (scinos) [reply packages/calypso-config/src/index.ts]: Good catch. I added the whitespace via search & replace in 2211b5ba48

File: apps/wpcom-block-editor/src/utils.js
Comment (jsnajdr) [inline apps/wpcom-block-editor/src/utils.js:71]: The warning can be fixed by referencing the `window` global instead:
```js
typeof window !== 'undefined' && window.navigator.userAgent
```

**Review by jsnajdr — APPROVED** [review]
:+1:

File: apps/wpcom-block-editor/src/utils.js
Comment (scinos) [reply apps/wpcom-block-editor/src/utils.js:71]: Fixed in https://github.com/Automattic/wp-calypso/pull/57013

**Pull request merged**
